modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryServiceTest.java

```diff
@@ -423,6 +495,50 @@ public void testDeleteObjectEntryHierarchy() throws Exception {
 			objectEntry -> Assert.assertNotNull(
 				_objectEntryService.deleteObjectEntry(
 					objectEntry.getObjectEntryId())));
+
+		// User can Delete bound entries if they can update root
+
+		_setUser(_adminUser);
+
+		objectEntryTree = TreeTestUtil.createObjectEntryTree(
+			"1", _objectEntryLocalService, _objectFieldLocalService,
+			_rootObjectDefinition.getRootObjectDefinitionId(),
+			_objectRelationshipLocalService, _treeFactory);
+
+		objectEntryRootNode = objectEntryTree.getRootNode();
+
+		_resourcePermissionLocalService.setResourcePermissions(
+			TestPropsValues.getCompanyId(),
+			_rootObjectDefinition.getClassName(),
+			ResourceConstants.SCOPE_INDIVIDUAL,
+			String.valueOf(objectEntryRootNode.getPrimaryKey()),
+			role.getRoleId(), new String[] {ActionKeys.UPDATE});
+
+		_setUser(_user);
+
+		TreeTestUtil.forEachNodeObjectEntry(
+			objectEntryTree.iterator(TreeConstants.ITERATOR_TYPE_POST_ORDER),
+			_objectEntryLocalService,
+			objectEntry -> {
+				if (objectEntry.getRootObjectEntryId() ==
+						objectEntry.getObjectEntryId()) {
+
+					return;
+				}
+
+				Assert.assertNotNull(
+					_objectEntryService.deleteObjectEntry(
+						objectEntry.getObjectEntryId()));
+			});
+
+		long rootObjectEntryId = objectEntryRootNode.getPrimaryKey();
+
+		AssertUtils.assertFailure(
+			PrincipalException.MustHavePermission.class,
+			StringBundler.concat(
+				"User ", _user.getUserId(), " must have DELETE permission for ",
+				_rootObjectDefinition.getClassName(), " ", rootObjectEntryId),
+			() -> _objectEntryService.deleteObjectEntry(rootObjectEntryId));
```

Comment:
What do you think about this comment?
//User cannot delete an object entry _of_ a root object definition with the update permission.